File: leetcode/removeNthNode.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
package leetcode

func removeNthFromEnd(head *ListNode, n int) *ListNode {
	if head == nil {
		return nil
	}

	list := &ListNode{0, head}
	ptr, run := list, head

	for i := 1; i <= n; i++ {
		run = run.Next
	}
	for run != nil {
		ptr = ptr.Next
		run = run.Next
	}
	ptr.Next = ptr.Next.Next
	return list.Next
}

File: leetcode/removeNthNode.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * @param {ListNode} head
 * @param {number} n
 * @return {ListNode}
 */
const removeNthFromEnd = function(head, n) {
    const holder = new ListNode(0)
    holder.next = head
    let first = holder
    let runner = holder
    
    for (let i = 1; i <= n+1; i++) {
        first = first.next
    }
    
    while (first !== null) {
        first = first.next
        runner = runner.next
    }
    runner.next = runner.next.next
    return holder.next
}
